src/main/java/redis/clients/jedis/JedisSentinelPool.java

```diff
@@ -315,7 +317,12 @@ public void onMessage(String channel, String message) {
 
         } catch (JedisException e) {
 
-          if (running.get()) {
+          try {
+            lock.unlock();
+          }catch (IllegalMonitorStateException ex){
```

Comment:
How can you get IllegalMonitorStateException? The lock is to be held by the current Thread at this stage.

Reply:
@gkorland Ideally the lock won't be held till this stage. It should be released at line 290.
We want to release the lock before starting to read from sentinel socket which is blocking call. 

This is necessary to be able to shutdown JedisSentinelPool or else if someone tries to shutdown, it will block on MasterListener thread, possibly for ever.